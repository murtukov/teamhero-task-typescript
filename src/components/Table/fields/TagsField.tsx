import React, {useContext} from 'react';
import {createUseStyles} from "react-jss";
import {IFieldProps} from "./sharedProps";
import {styles as sharedStyles} from "./styles";
import {TableContext} from "../TableProvider";

export interface ITagsFieldProps extends IFieldProps {
    data?: string[]
}

function TagsField({data, source}: ITagsFieldProps) {
    const c = useStyles();
    const {filterOptions, setFilterOptions} = useContext(TableContext);

    function handleClick(tagName: string) {
        setFilterOptions({
            column: source,
            tags: [...filterOptions.tags, tagName]
        })
    }

    if (undefined === data) {
        return null;
    }

    return (
        <td className={c.root}>
            {data.map((tag, i) =>
                <span className={c.tag} onClick={() => handleClick(tag)} key={i}>
                    {tag}
                </span>
            )}
        </td>
    );
}

const useStyles = createUseStyles({
    ...sharedStyles,
    tag: {
        backgroundColor: 'rgb(254, 240, 216)',
        padding: 5,
        borderRadius: 5,
        marginRight: 7,
        cursor: 'pointer'
    },
});

export default TagsField;