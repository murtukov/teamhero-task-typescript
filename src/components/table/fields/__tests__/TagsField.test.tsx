import React from "react";
import {render} from "@testing-library/react";
import TagsField from "../TagsField";

const tags = ['One', 'Two', 'Three'];

describe('TagsField component', () => {
    it('matches snapshot', () => {
        const { container } = render(<TagsField source='dummy' data={tags}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('contains correct amount of tags', () => {
        const { container, queryByText } = render(<TagsField source='dummy' data={tags}/>);

        expect(container.firstChild!.childNodes.length).toBe(3);
        expect(queryByText('One')).toBeInTheDocument();
        expect(queryByText('Two')).toBeInTheDocument();
        expect(queryByText('Three')).toBeInTheDocument();
    });
})