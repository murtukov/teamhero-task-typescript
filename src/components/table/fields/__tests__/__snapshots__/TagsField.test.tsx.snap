// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsField component renders correctly 1`] = `
<div
  class="cell-0-2-1 cell-d0-0-2-3"
>
  <span
    class="tag-0-2-2"
  >
    One
  </span>
  <span
    class="tag-0-2-2"
  >
    Two
  </span>
  <span
    class="tag-0-2-2"
  >
    Three
  </span>
</div>
`;
